### PR TITLE
canonicalize.ensure_bool_predicates

### PR DESCRIPTION
**Pull request by barakalon**
query:
```sql
SELECT
  a
FROM (
  SELECT 1 AS a
) AS x
WHERE a
```

Postgres:
```
error: argument of WHERE must be type boolean, not type integer
```

MySQL:
```
a
1
```

### CODE REVIEW

File: sqlglot/optimizer/canonicalize.py [hunk 1]
```diff
@@ -52,6 +53,17 @@ def remove_redundant_casts(expression: exp.Expression) -> exp.Expression:
     return expression
 
 
+def ensure_bool_predicates(expression: exp.Expression) -> exp.Expression:
+    if isinstance(expression, exp.Connector):
+        _replace_int_predicate(expression.left)
+        _replace_int_predicate(expression.right)
+
+    elif isinstance(expression, (exp.Where, exp.Having)):
```
Comment (GeorgeSittas) [inline sqlglot/optimizer/canonicalize.py:61]: Should we consider `exp.Qualify` as well?

**Review by GeorgeSittas — APPROVED** [review]

File: sqlglot/optimizer/canonicalize.py [hunk 2]
```diff
@@ -68,3 +80,8 @@ def _replace_cast(node: exp.Expression, to: str) -> None:
     cast = exp.Cast(this=node.copy(), to=data_type)
     cast.type = data_type
     node.replace(cast)
+
+
+def _replace_int_predicate(expression: exp.Expression) -> None:
+    if expression.type and expression.type.this in exp.DataType.INTEGER_TYPES:
+        expression.replace(exp.GT(this=expression.copy(), expression=exp.Literal.number(0)))
```
Comment (tobymao) [inline sqlglot/optimizer/canonicalize.py:87]: are negative types falsey?

File: sqlglot/optimizer/canonicalize.py (hunk 1 above)
Comment (barakalon) [reply sqlglot/optimizer/canonicalize.py:61]: I don't know much about QUALIFY. Do we annotate all the window functions properly?

File: sqlglot/optimizer/canonicalize.py (hunk 2 above)
Comment (barakalon) [reply sqlglot/optimizer/canonicalize.py]: Switched to NEQ

File: sqlglot/optimizer/canonicalize.py (hunk 1 above)
Comment (GeorgeSittas) [reply sqlglot/optimizer/canonicalize.py:61]: Good point, I'm not sure yet. We can tackle this in a future PR then.

**Pull request merged**